src/tests/GeoTest.cpp

```diff
@@ -98,3 +99,64 @@ TEST(GeoTest, CoordinateFrame) {
   c4.fromJson(j);
   EXPECT_EQ(c3, c4);
 }
+
+// standard method
+// transform the 8 corners, and extract the min and max
+Magnum::Range3D getTransformedBB_standard(const Magnum::Range3D& range,
+                                          const Magnum::Matrix4& xform) {
+  std::vector<Magnum::Vector3> corners;
+  corners.push_back(xform.transformPoint(range.frontBottomLeft()));
+  corners.push_back(xform.transformPoint(range.frontBottomRight()));
+  corners.push_back(xform.transformPoint(range.frontTopLeft()));
+  corners.push_back(xform.transformPoint(range.frontTopRight()));
+
+  corners.push_back(xform.transformPoint(range.backTopLeft()));
+  corners.push_back(xform.transformPoint(range.backTopRight()));
+  corners.push_back(xform.transformPoint(range.backBottomLeft()));
+  corners.push_back(xform.transformPoint(range.backBottomRight()));
+
+  Magnum::Range3D transformedBB{Magnum::Math::minmax<Magnum::Vector3>(corners)};
+
+  return transformedBB;
+}
+
+TEST(GeoTest, Aabb) {
+  // Generate N boxes in random position and orientation.
+  // compute aabb for each box using standard method and library method
+  // respectively.
+  // compare the results, which should be identical
+
+  Magnum::Range3D box = Magnum::Range3D{Magnum::Vector3{-10.0, -10.0, -10.0},
+                                        Magnum::Vector3{10.0, 10.0, 10.0}};
+
+  int numBoxes = 100;
+  for (int iBox = 0; iBox < numBoxes; ++iBox) {
+    // generate random translation
+    const Magnum::Vector3 translation{
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f,
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f,
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f};
+
+    // generate random rotation using:
+    // http://planning.cs.uiuc.edu/node198.html
+    double u1 = (rand() % 1000) / 1000.0;
+    double u2 = (rand() % 1000) / 1000.0;
+    double u3 = (rand() % 1000) / 1000.0;
+
+    Magnum::Vector3 qAxis(sqrt(1 - u1) * cos(2 * M_PI * u2),
+                          sqrt(u1) * sin(2 * M_PI * u3),
+                          sqrt(u1) * cos(2 * M_PI * u3));
+
+    // assemble the transformation matrix
+    Magnum::Matrix4 xform = Magnum::Matrix4::from(
+        Magnum::Quaternion(qAxis, sqrt(1 - u1) * sin(2 * M_PI * u2)).toMatrix(),
+        translation);
+
+    Magnum::Range3D aabbControl = getTransformedBB_standard(box, xform);
+    Magnum::Range3D aabbTest = esp::geo::getTransformedBB(box, xform);
+
+    float eps = 1e-12;
+    EXPECT_LE((aabbControl.min() - aabbTest.min()).dot(), eps);
+    EXPECT_LE((aabbControl.max() - aabbTest.max()).dot(), eps);
```

Comment:
Like I did with `SimTest` in #496, maybe consider rewriting this test using `Corrade::TestSuite` as well? It should be rather easy I think. The above could then be turned into what's below, printing a much better diagnostic in case of comparison failure:

```cpp
CORRADE_COMPARE_AS(aabbTest, aabbControl, 
    Cr::TestSuite::Compare::around(1e-13f));
```

Reply:
OK, will do.

Reply:
@mosra:

Cannot use what you suggested. The error message reads:
```
n file included from /home/user/habitat-sim/src/tests/AabbTest.cpp:5:                                                 
/home/user/habitat-sim/src/tests/AabbTest.cpp: In member function ‘void {anonymous}::GeoTest::aabb()’:                
/home/user/habitat-sim/src/tests/AabbTest.cpp:78:54: error: temporary of non-literal type ‘Corrade::TestSuite::Compare
::Around<float>’ in a constant expression                                                                             
   78 |                        Cr::TestSuite::Compare::around(eps));                                                  
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~                                                    
/home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Tester.h:1377:27: note: in definition of macro ‘CORRADE_
COMPARE_AS’                                                                                                           
 1377 |         Tester::compareAs<__VA_ARGS__>(#actual, actual, #expected, expected); \                               
      |                           ^~~~~~~~~~~                                                                         
In file included from /home/user/habitat-sim/src/tests/AabbTest.cpp:6:                                                
/home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Compare/Numeric.h:108:25: note: ‘Corrade::TestSuite::Com
pare::Around<float>’ is not literal because:                                                                          
  108 | template<class T> class Around {                                                                              
      |                         ^~~~~~                                                                                
/home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Compare/Numeric.h:108:25: note:   ‘Corrade::TestSuite::C
ompare::Around<float>’ is not an aggregate, does not have a trivial default constructor, and has no ‘constexpr’ constr
uctor that is not a copy or move constructor                                                                          
In file included from /home/user/habitat-sim/src/tests/AabbTest.cpp:5:                                                
/home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Tester.h:1377:76: error: no matching function for call t
o ‘{anonymous}::GeoTest::compareAs<Corrade::TestSuite::Compare::around<float>(eps)>(const char [9], Magnum::Range3D&, 
const char [12], Magnum::Range3D&)’                                                                                   
 1377 |         Tester::compareAs<__VA_ARGS__>(#actual, actual, #expected, expected); \                               
      |                                                                            ^                                  
/home/user/habitat-sim/src/tests/AabbTest.cpp:77:5: note: in expansion of macro ‘CORRADE_COMPARE_AS’                  
   77 |     CORRADE_COMPARE_AS(aabbTest, aabbControl,                                                                       |     ^~~~~~~~~~~~~~~~~~                                                                                        
/home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Tester.h:1123:78: note: candidate: ‘template<template<cl
ass> class T, class Actual, class Expected> void Corrade::TestSuite::Tester::compareAs(const char*, const Actual&, const char*, const Expected&)’                                                                                           
 1123 |         template<template<class> class T, class Actual, class Expected> void compareAs(const char* actual, const Actual& actualValue, const char* expected, const Expected& expectedValue) {                                        
      |                                                                              ^~~~~~~~~                        /home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Tester.h:1123:78: note:   template argument deduction/su
bstitution failed:                                                                                                    /home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Tester.h:1128:50: note: candidate: ‘template<class T, cl
ass U, class V> void Corrade::TestSuite::Tester::compareAs(const char*, const U&, const char*, const V&)’             
 1128 |         template<class T, class U, class V> void compareAs(const char* actual, const U& actualValue, const char* expected, const V& expectedValue) {                                                                                
      |                                                  ^~~~~~~~~                                                    /home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Tester.h:1128:50: note:   template argument deduction/substitution failed:    
```

Reply:
Argh, sorry, should have been `WITH`, not `AS`, and comparing the two vectors instead of the whole Range:

```cpp
CORRADE_COMPARE__WITH(aabbTest.min(), aabbControl.min(), 
    Cr::TestSuite::Compare::around(1e-12f));
CORRADE_COMPARE__WITH(aabbTest.max(), aabbControl.max(), 
    Cr::TestSuite::Compare::around(1e-12f));
```

Btw., I think `1e-12f` is too little and could lead to failures depending on compiler / optimization level, `1e-7f` or `1e-6f` could be not overly strict but still precise enough.

Reply:
OK, It depends if I compare the distance or the square distance. Usually I use 1e-6f for the distance and 1e-10f ~1e-12f for the square distance.

Your comments make sense. I can change.

Reply:
hmmm.... Still not correct.

```
/home/user/habitat-sim/src/tests/AabbTest.cpp:77:5:   required from here                                              
/home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Tester.h:1516:129: error: invalid initialization of reference of type ‘const ActualType&’ {aka ‘const float&’} from expression of type ‘const Magnum::Math::Vector3<float>’   
 1516 |     const typename Implementation::ComparatorTraits<T, typename std::decay<U>::type, typename std::decay<V>::type>::ActualType& actualValueInExpectedActualType = actualValue;                                                      
            |                                                                                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                               
              /home/user/habitat-sim/src/deps/corrade/src/Corrade/TestSuite/Tester.h:1517:131: error: invalid initialization of reference of type ‘const ExpectedType&’ {aka ‘const float&’} from expression of type ‘const Magnum::Math::Vector3<float>’                                                                                                         
               1517 |     const typename Implementation::ComparatorTraits<T, typename std::decay<U>::type, typename std::decay<V>::type>::ExpectedType& expectedValueInExpectedExpectedType = expectedValue;                                
                  |                                                                                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                   
                    gmake[2]: *** [tests/CMakeFiles/AabbTest.dir/build.make:63: tests/CMakeFiles/AabbTest.dir/AabbTest.cpp.o] Error 1                                                                                                       
                    gmake[1]: *** [CMakeFiles/Makefile2:6263: tests/CMakeFiles/AabbTest.dir/all] Error 2 
```

Reply:
Okay, then this:

```cpp
CORRADE_COMPARE__WITH(aabbTest.min(), aabbControl.min(), 
    Cr::TestSuite::Compare::around(Mn::Vector3{1e-12f}));
CORRADE_COMPARE__WITH(aabbTest.max(), aabbControl.max(), 
    Cr::TestSuite::Compare::around(Mn::Vector3{1e-12f}));
```